org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/internal/SourceFolderUpdater.java

```diff
@@ -68,19 +69,40 @@ private void updateClasspath(IProgressMonitor monitor) throws JavaModelException
         updateClasspath(newClasspathEntries, monitor);
     }
 
-    private List<IClasspathEntry> collectGradleSourceFolders() {
+    private List<IClasspathEntry> collectGradleSourceFolders() throws JavaModelException {
+        final List<IClasspathEntry> rawClasspath = ImmutableList.copyOf(this.project.getRawClasspath());
         List<IClasspathEntry> sourceFolders = FluentIterable.from(this.sourceFolders).transform(new Function<OmniEclipseSourceDirectory, IClasspathEntry>() {
 
             @Override
             public IClasspathEntry apply(OmniEclipseSourceDirectory directory) {
                 IFolder sourceDirectory = SourceFolderUpdater.this.project.getProject().getFolder(Path.fromOSString(directory.getPath()));
                 FileUtils.ensureFolderHierarchyExists(sourceDirectory);
-                IPackageFragmentRoot root = SourceFolderUpdater.this.project.getPackageFragmentRoot(sourceDirectory);
+                final IPackageFragmentRoot root = SourceFolderUpdater.this.project.getPackageFragmentRoot(sourceDirectory);
                 IClasspathAttribute fromGradleModel = JavaCore.newClasspathAttribute(CLASSPATH_ATTRIBUTE_FROM_GRADLE_MODEL, "true");
+                
+                IPath[] includes = new IPath[] {};
+                IPath[] excludes = new IPath[] {};
```

Comment:
It can be made a bit more compact: 
IPath[] includes = userDefinedClasspathEntry.isPresent() ? userDefinedClasspathEntry.get().getInclusionPatterns() : new IPath[] {};
                IPath[] excludes = userDefinedClasspathEntry.isPresent() ? userDefinedClasspathEntry.get().getExclusionPatterns() : new IPath[] {};